codalab/common.py

```diff
@@ -272,25 +288,30 @@ def _get_signed_url(self, path, **kwargs):
             "/", 1
         )
         bucket = client.get_bucket(bucket_name)
-        blob = bucket.get_blob(blob_name)
+        blob = bucket.blob(blob_name)
         signed_url = blob.generate_signed_url(
             version="v4",
             expiration=datetime.timedelta(hours=1),
-            method="GET",
-            response_disposition=kwargs["content_disposition"],
-            response_type=kwargs["content_type"],
+            method=kwargs.get("method", "GET"),  # HTTP method. eg, GET, PUT
+            # TODO: test what if kwargs does not have this keyword.
+            content_type=kwargs.get("request_content_type", None),
+            response_disposition=kwargs.get("content_disposition", None),
+            response_type=kwargs.get("content_type", None),
         )
         return signed_url
 
-    def bundle_path_sas_url(self, **kwargs):
-        return self._get_sas_url(self.bundle_path, **kwargs)
+    def bundle_path_sas_url(self, permission='r', **kwargs):
```

Comment:
We can remove this and just do 

```
def bundle_path_sas_url(self, **kwargs):
    return self._get_sas_url(self.bundle_path, **kwargs)
```

Because permission is a kwarg